src/components/pages/__tests__/EktefelleStep.test.tsx

```diff
@@ -0,0 +1,177 @@
+import { render, screen, fireEvent } from '@testing-library/react'
+import EktefelleStep from '../EktefelleStep'
+import { FormContext } from '@/contexts/context'
+import useErrorHandling from '../../../helpers/useErrorHandling'
+import { State } from '@/common'
+import { initialFormState } from '@/defaults/defaultFormState'
+import { useFieldChange } from '@/helpers/useFormState'
+
+// Mock the useErrorHandling hook
+jest.mock('../../../helpers/useErrorHandling', () => ({
+  __esModule: true,
+  default: jest.fn(),
+}))
+
+// Mock the FormButtons component
+jest.mock('../../FormButtons', () => ({
+  __esModule: true,
+  default: jest.fn(() => <div>Mocked FormButtons</div>),
+}))
+
+// Mock the useFieldChange hook
+jest.mock('@/helpers/useFormState', () => ({
+  __esModule: true,
+  useFieldChange: jest.fn(),
+}))
+
+const mockGoToNext = jest.fn()
+const mockSetState = jest.fn()
+const mockHandleFieldChange = jest.fn((updateFn) => {
+  const draft: Partial<State> = {}
+  updateFn(draft)
+  return draft
+})
+
+const defaultFormPageProps = {
+  curStep: 1,
+  length: 5,
+  goBack: jest.fn(),
+  onStepChange: jest.fn(),
+  handleSubmit: jest.fn(),
+  goToNext: mockGoToNext,
+}
+
+const context = {
+  setState: mockSetState,
+  state: initialFormState,
+  formPageProps: defaultFormPageProps,
+}
+
+const mockValidateFields = jest.fn()
+const mockClearError = jest.fn()
+
+beforeEach(() => {
+  jest.clearAllMocks()
+  ;(useErrorHandling as jest.Mock).mockReturnValue([
+    {},
+    { validateFields: mockValidateFields, clearError: mockClearError },
+  ])
+  ;(useFieldChange as jest.Mock).mockReturnValue({
+    handleFieldChange: mockHandleFieldChange,
+  })
+})
+
+const renderComponent = (contextOverride = context) => {
+  return render(
+    <FormContext.Provider value={contextOverride}>
+      <EktefelleStep grunnbelop={100000} />
+    </FormContext.Provider>
+  )
+}
+
+describe('EktefelleStep Component', () => {
+  test('Burde rendre komponenten', () => {
+    renderComponent()
+    expect(screen.getByLabelText('Hva er din sivilstand?')).toBeInTheDocument()
+    expect(screen.getByText('Mocked FormButtons')).toBeInTheDocument()
+  })
+
+  test('Burde gå videre til neste step når skjemaet valideres uten feil', () => {
+    mockValidateFields.mockReturnValue(false)
+    renderComponent()
+    const form = screen.getByRole('form')
+    fireEvent.submit(form)
+    expect(mockValidateFields).toHaveBeenCalledWith('EktefelleStep')
+    expect(mockGoToNext).toHaveBeenCalled()
+  })
+
+  test('Burde ikke gå videre til neste step når skjemaet valideres med feil', () => {
+    mockValidateFields.mockReturnValue(true)
+    renderComponent()
+    const form = screen.getByRole('form')
+    fireEvent.submit(form)
+    expect(mockValidateFields).toHaveBeenCalledWith('EktefelleStep')
+    expect(mockGoToNext).not.toHaveBeenCalled()
+  })
+
+  describe('sivilstand', () => {
```

Comment:
Usikker på tanken med denne korte beskrivelsene. Husk at det skal gi mening når testene kjører. Kunne det heller være "Gitt at brukeren har valgt sivilstand, "... og senere "Når den er satt på "Ugift"," ... "Burde to radioknapper vises."